cms/djangoapps/contentstore/views/tests/test_import_export.py

```diff
@@ -241,6 +247,81 @@ def try_tar(tarpath):
         import_status = json.loads(resp_status.content)["ImportStatus"]
         self.assertIn(import_status, (0, 3))
 
+    def test_library_import(self):
+        """
+        Try importing a known good library archive, and verify that the
+        contents of the library have completely replaced the old contents.
+        """
+        # Create some blocks to overwrite
+        library = LibraryFactory.create(modulestore=self.store)
+        lib_key = library.location.library_key
+        test_block = ItemFactory.create(
+            category="vertical",
+            parent_location=library.location,
+            user_id=self.user.id,
+            publish_item=False,
+        )
+        test_block2 = ItemFactory.create(
+            category="vertical",
+            parent_location=library.location,
+            user_id=self.user.id,
+            publish_item=False
+        )
+        # Create a library and blocks that should remain unmolested.
+        unchanged_lib = LibraryFactory.create()
+        unchanged_key = unchanged_lib.location.library_key
+        test_block3 = ItemFactory.create(
+            category="vertical",
+            parent_location=unchanged_lib.location,
+            user_id=self.user.id,
+            publish_item=False
+        )
+        test_block4 = ItemFactory.create(
+            category="vertical",
+            parent_location=unchanged_lib.location,
+            user_id=self.user.id,
+            publish_item=False
+        )
+        # Refresh library.
+        library = self.store.get_library(lib_key)
+        children = [self.store.get_item(child).url_name for child in library.children]
+        self.assertEqual(len(children), 2)
+        self.assertIn(test_block.url_name, children)
+        self.assertIn(test_block2.url_name, children)
+
+        unchanged_lib = self.store.get_library(unchanged_key)
+        children = [self.store.get_item(child).url_name for child in unchanged_lib.children]
+        self.assertEqual(len(children), 2)
+        self.assertIn(test_block3.url_name, children)
+        self.assertIn(test_block4.url_name, children)
+
+        extract_dir = path(tempfile.mkdtemp())
+        try:
+            tar = tarfile.open(path(TEST_DATA_DIR) / 'uploads' / 'library.HhJfPD.tar.gz')
+            safetar_extractall(tar, extract_dir)
+            library_items = import_library_from_xml(
```

Comment:
@cahrens As far as I've been able to tell, full imports/exports don't quite happen through those other tests in `common`, but some unit tests are there, mostly for XModules, which have their own methods for import and export. You'll notice that full import/exports happen here, and that's because some runtime info is not set in the common tests.